src/uUV.cpp

```diff
@@ -1,11 +1,356 @@
+}
+
+void uv_async_init(uv_loop_t *loop, uv_async_t *async, uv_async_cb cb) {
+    async->loopIndex = loop->index;
+    loop->numEvents++;
+
+    async->cbIndex = asyncCbHead;
+    for (int i = 0; i < asyncCbHead; i++) {
+        if (async_callbacks[i] == cb) {
+            async->cbIndex = i;
+            break;
+        }
+    }
+    if (async->cbIndex == asyncCbHead) {
+        async_callbacks[asyncCbHead++] = cb;
+    }
+
+    loop->asyncs.insert(async);
+}
+
+void uv_async_send(uv_async_t *async) {
+    uv_loop_t *loop = async->get_loop();
+    loop->async_mutex.lock();
+    uint64_t val = 1;
+    int w = write(loop->asyncWakeupFd, &val, sizeof(val));
+    async->run = true;
+    loop->async_mutex.unlock();
+}
+
+void uv_close(uv_async_t *handle, uv_close_cb cb) {
+    uv_loop_t *loop = handle->get_loop();
+
+    loop->asyncs.erase((uv_async_t *) handle);
+
+    handle->flags |= UV_HANDLE_CLOSING;
+    loop->closing.push_back({(uv_handle_t *) handle, cb});
+}
+
+void uv_idle_init(uv_loop_t *loop, uv_idle_t *idle) {
+    idle->loopIndex = loop->index;
+    loop->numEvents++;
+}
+
+void uv_idle_start(uv_idle_t *idle, uv_idle_cb cb) {
+    idle->cbIndex = idleCbHead;
+    for (int i = 0; i < idleCbHead; i++) {
+        if (idle_callbacks[i] == cb) {
+            idle->cbIndex = i;
+            break;
+        }
+    }
+    if (idle->cbIndex == idleCbHead) {
+        idle_callbacks[idleCbHead++] = cb;
+    }
+
+    idle->get_loop()->idlers.insert(idle);
+}
+
+void uv_idle_stop(uv_idle_t *idle) {
+    idle->get_loop()->idlers.erase(idle);
+}
+
+void uv_close(uv_idle_t *handle, uv_close_cb cb) {
+    uv_loop_t *loop = handle->get_loop();
+    handle->flags |= UV_HANDLE_CLOSING;
+    loop->closing.push_back({(uv_handle_t *) handle, cb});
+}
+
+uv_poll_cb uv_poll_t::get_poll_cb() const {
+    return poll_callbacks[cbIndex];
+}
+
+int uv_poll_init_socket(uv_loop_t *loop, uv_poll_t *poll, uv_os_sock_t socket) {
+    int flags = fcntl(socket, F_GETFL, 0);
+    if (flags == -1) {
+        return -1;
+    }
+    flags |= O_NONBLOCK;
+    flags = fcntl (socket, F_SETFL, flags);
+    if (flags == -1) {
+        return -1;
+    }
+
+    poll->loopIndex = loop->index;
+    poll->fd = socket;
+    poll->event.events = 0;
+    poll->event.data.ptr = poll;
+    loop->numEvents++;
+    return epoll_ctl(loop->efd, EPOLL_CTL_ADD, socket, &poll->event);
+}
+
+int uv_poll_start(uv_poll_t *poll, int events, uv_poll_cb cb) {
+    poll->flags |= UV_HANDLE_RUNNING;
+    poll->event.events = events;
+    poll->cbIndex = pollCbHead;
+    for (int i = 0; i < pollCbHead; i++) {
+        if (poll_callbacks[i] == cb) {
+            poll->cbIndex = i;
+            break;
+        }
+    }
+    if (poll->cbIndex == pollCbHead) {
+        poll_callbacks[pollCbHead++] = cb;
+    }
+    return epoll_ctl(poll->get_loop()->efd, EPOLL_CTL_MOD, poll->fd, &poll->event);
+}
+
+int uv_poll_stop(uv_poll_t *poll) {
+    poll->flags &= ~UV_HANDLE_RUNNING;
```

Comment:
Actually, these edge cases are rare enough / shouldn't occur in  our lib that I think I'll just remove this stuff for now. It can easily be added later if there are issues.